zebra-state/tests/basic.rs

```diff
@@ -0,0 +1,96 @@
+use color_eyre::eyre::Report;
+use color_eyre::eyre::{bail, ensure, eyre};
+use once_cell::sync::Lazy;
+use std::sync::{Arc, Once};
+use tempdir::TempDir;
+use tower::Service;
+use tracing_error::ErrorLayer;
+use tracing_subscriber::prelude::*;
+use tracing_subscriber::{fmt, EnvFilter};
+use zebra_chain::{
+    block::{Block, BlockHeaderHash},
+    serialization::ZcashDeserialize,
+};
+use zebra_test::transcript::Transcript;
+
+use zebra_state::*;
+
+static LOGGER_INIT: Once = Once::new();
+
+fn install_tracing() {
+    LOGGER_INIT.call_once(|| {
+        let fmt_layer = fmt::layer().with_target(false);
+        let filter_layer = EnvFilter::try_from_default_env()
+            .or_else(|_| EnvFilter::try_new("info"))
+            .unwrap();
+
+        tracing_subscriber::registry()
+            .with(filter_layer)
+            .with(fmt_layer)
+            .with(ErrorLayer::default())
+            .init();
+    })
+}
+
+static ADD_BLOCK_TRANSCRIPT: Lazy<Vec<(Request, Response)>> = Lazy::new(|| {
+    let block: Arc<_> =
+        Block::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_415000_BYTES[..])
+            .unwrap()
+            .into();
+    let hash = block.as_ref().into();
+    vec![
+        (
+            Request::AddBlock {
+                block: block.clone(),
+            },
+            Response::Added { hash },
+        ),
+        (Request::GetBlock { hash }, Response::Block { block }),
+    ]
+});
+
+static GET_TIP_TRANSCRIPT: Lazy<Vec<(Request, Response)>> = Lazy::new(|| {
+    let block0: Arc<_> =
+        Block::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_GENESIS_BYTES[..])
+            .unwrap()
+            .into();
+    let block1: Arc<_> = Block::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_1_BYTES[..])
+        .unwrap()
+        .into();
+    let hash0 = block0.as_ref().into();
+    let hash1 = block1.as_ref().into();
+    vec![
+        // Insert higher block first, lower block second
+        (
+            Request::AddBlock { block: block1 },
+            Response::Added { hash: hash1 },
+        ),
+        (
+            Request::AddBlock { block: block0 },
+            Response::Added { hash: hash0 },
+        ),
+        (Request::GetTip, Response::Tip { hash: hash1 }),
+    ]
+});
+
+#[tokio::test]
+async fn check_transcripts() -> Result<(), Report> {
```

Comment:
I think it might be a good idea to have individual tests for each transcript so they can be run independently

Reply:
I merged it all into one test to cut down on the common code so that it's very easy to extend the list of transcripts.  I guess the downside is that they can't be run independently, but because we're using `Report` we should get isolated errors on any failures, and I'm not sure how often we'd want to run individual transcripts?

Reply:
Once we have a lot of tests I expect if one starts failing we will want to be able to run `cargo test -- relevant_test_name`, but if that comes up we can move the important transcript to its own test, there's no real reason to do it now.